bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -3202,6 +3202,16 @@ void ReplicaImp::onViewsChangeTimer(Timers::Handle timer)  // TODO(GG): review/u
     const uint64_t diffMilli2 = duration_cast<milliseconds>(currTime - timeOfLastViewEntrance).count();
     const uint64_t diffMilli3 = duration_cast<milliseconds>(currTime - timeOfEarliestPendingRequest).count();
 
+    const auto delayedCIDs = clientsManager->getAllPendingRequestsExceedingThreshold(viewChangeTimeout / 2, currTime);
```

Comment:
Why loop over all requests again? You can log the massage directly in ClientsManager::getAllPendingRequestsExceedingThreshold.

Reply:
You are right, since we only need them for logging there is no need. Will make the changes, thanks @yuliasherman .